mrsal/amqp/subclass.py

```diff
@@ -0,0 +1,234 @@
+import pika
+from ssl import SSLContext
+from mrsal.exceptions import MrsalAbortedSetup
+from logging import WARNING
+from pika.exceptions import AMQPConnectionError, ChannelClosedByBroker, StreamLostError, ConnectionClosedByBroker
+from pika.adapters.asyncio_connection import AsyncioConnection
+from typing import Callable, Type
+from tenacity import retry, stop_after_attempt, wait_fixed, retry_if_exception_type, before_sleep_log
+from pydantic import ValidationError
+from pydantic.dataclasses import dataclass
+from neolibrary.monitoring.logger import NeoLogger
+
+from mrsal.superclass import Mrsal
+from pydantic.deprecated.tools import json
+
+log = NeoLogger(__name__, rotate_days=10)
```

Comment:
Isn't it juicy to be able to choose rotation days from env?

Reply:
Indeed